### PR TITLE
Budget App Milestone

### PR DESCRIPTION
**Pull request by Nejjari001**
## Live Demo Link: 
https://kango-budget-app.onrender.com/

## Video explanation:
https://www.loom.com/share/2bf62dcc2e90464a8b46fe17f0744fb1

- [ ] Splash screen
- A simple page with the name of your app (yes, you need to choose one), and links to the signup and log-in pages.

- [ ] Sign-up and login pages
- The user should be able to register in the app with full name, email, and password (all mandatory).
- The user can log into the app using email and password.
- If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- [ ] Home page (categories page)
- When the user logs in, they are presented with the categories page.
- For each category, the user can see their name, icon, and the total amount of all the transactions that belong to that category.
- When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
- There is a button "add a new category" at the bottom that brings the user to the page to create a new category

- [ ] Transactions page
- For a given category, the list of transactions is presented, ordered by the most recent.
- At the top of the page the user can see the total amount for the category (the sum of all of the amounts of the transactions in that category).
- There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
- When the user clicks on the "Back" button (<), the user navigates to the home page.

- [ ] "Add a new category" page
-  The user fills out a form to create a new category, indicating their name and icon (both mandatory).
- The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
- When the user clicks on the "Back" button (<), the user navigates to the home page.

- [ ] "Add a new transaction" page
- The user fills out a form to create a new transaction with the following:
name (mandatory), amount (mandatory), categories (mandatory at least one)
- The user clicks (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
- When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

- [ ] Testing requirements
- Create unit and integration tests for all the essential components of your RoR application.

### CODE REVIEW

File: app/views/groups/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+          </div>
+        <% end %>
+    <% end %>
+    <div class="add-category">
+      <%= link_to 'Add Category', new_user_group_path, class: "" %>
+    </div>
+  </div>
+
+
+</div>
```
Comment (thecodechaser) [inline app/views/groups/index.html.erb]: - [ ] Please make sure the user can see the total amount of all the transactions that belong to that category on the category(home) page as requested in the project requirements. Currently, The name and icon are visible but not the total amount of all the transactions.

image

File: app/views/entities/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+      <p class="">ADD TRANSACTION</p>
+      <%= image_tag("search.png", :alt => "Search") %>
+    </div>
+  </div>
+
+  <%= render "form", entity: @entity %>
+  <div class="back-to-groups">
+    <%= link_to 'Back to transactions', user_groups_path(current_user) %>
+  </div>
+</div>
```
Comment (thecodechaser) [inline app/views/entities/new.html.erb]: - [ ] Please make sure when the user clicks the "Save" button to create the new transaction user is redirected to the transactions page for that category as requested in the project requirements. Currently, the User is redirecting to the category(home) page.

image

Comment (thecodechaser) [inline app/views/entities/new.html.erb]: - [ ] Please make sure when the user clicks on the "Back" button on the Add a new transaction page the user is redirected to the transactions page for that category as requested in the project requirements. Currently, the User is redirecting to the category(home) page.

image

File: README.md [hunk 3]
```diff
@@ -1,24 +1,215 @@
...
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+- [Live Demo Link](https://rails-3xzd.onrender.com)
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+
```
Comment (thecodechaser) [inline README.md]: - [ ] The live demo link in the README is not working, There are some errors please fix them and make sure the project is deployed and accessible online and the live demo link in the README works as expected as requested in the project requirements.

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @Nejjari001👋, 

### You are doing a marvelous job on the RoR capstone project and this is one of the best projects I have seen so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

image

## To Highlight 🎉
- All tests are passed. ✔️
- Highly organized code. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- Nice UI design. ✔️
- PR has a good title and summary. ✔️



## Required Changes ♻️

- [ ] Please make sure to record a video presenting the features of the project you built. It is a video with a maximum length of 5 minutes as requested in the project requirements. Add the link to the presentation video in the PR and in the README file. The content of the video should include:
  - Description of the project.
  - Demo of the project features.
  - You should also highlight some interesting piece of code.

- [ ] Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/groups/index.html.erb (hunk 1 above)
Comment (Whoistolu) [reply app/views/groups/index.html.erb]: - Well resolved 👍🏽 

Screenshot 2023-03-18 at 17 20 57

File: app/views/entities/new.html.erb (hunk 2 above)
Comment (Whoistolu) [reply app/views/entities/new.html.erb]: - Well resolved 👍🏽

Comment (Whoistolu) [reply app/views/entities/new.html.erb]: - Well resolved 👍🏽

File: README.md (hunk 3 above)
Comment (Whoistolu) [reply README.md]: - Well resolved 👍🏽 
Screenshot 2023-03-18 at 17 17 37

**Review by Whoistolu — APPROVED** [review]
### Hi @Nejjari001 

Great job implementing all the recommendations by the previous reviewer 👍🏽 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**